api/src/test/java/org/openmrs/module/ModuleUtilTest.java

```diff
@@ -651,4 +652,131 @@ public void getResourceFromApi_shouldReturnNullIfFileIsNotFoundInApi() throws Ex
 		Assert.assertNull(resultStream);
 	}
+	
+	/**
+	 * @see ModuleUtil#expandJar(File,File,String,boolean)
+	 */
+	@Test
+	@Verifies(value = "expand entire jar if name is null", method = "expandJar(File,File,String,boolean)")
+	public void expandJar_shouldExpandEntireJarIfNameIsNull() throws IOException {
+		// Arrange
+		final int numberOfFilesInJar = 6;
+		File destinationFolder = this.getEmptyJarDestinationFolder();
+		// Act
+		ModuleUtil.expandJar(getJarFile(), destinationFolder, null, false);
+		// Assert
+		Assert.assertEquals(numberOfFilesInJar, FileUtils.listFiles(destinationFolder, null, true).size());
+		// Clean up
+		FileUtils.deleteDirectory(destinationFolder);
+	}
+	
+	/**
+	 * @see ModuleUtil#expandJar(File,File,String,boolean)
+	 */
+	@Test
+	@Verifies(value = "expand directory with parent tree if name is directory and keepFullPath is true", method = "expandJar(File,File,String,boolean)")
+	public void expandJar_shouldExpandDirectoryWithParentTreeIfNameIsDirectoryAndKeepFullPathIsTrue() throws IOException {
+		// Arrange
+		final int numberOfFilesInSpecifiedJarDirectory = 2;
+		String directoryPath = "META-INF/maven/org.openmrs.module/test1-api";
+		File destinationFolder = this.getEmptyJarDestinationFolder();
+		// Act
+		ModuleUtil.expandJar(getJarFile(), destinationFolder, directoryPath, true);
+		// Assert
+		Object[] filesList = FileUtils.listFiles(destinationFolder, null, true).toArray();
+		Assert.assertEquals(numberOfFilesInSpecifiedJarDirectory, filesList.length);
+		String expectedPath = destinationFolder.getPath() + File.separator + directoryPath.replace("/", File.separator) + File.separator;
+		String ActualPath = ((File)filesList[0]).getPath().replace(((File)filesList[0]).getName(), "");
+		Assert.assertEquals(expectedPath, ActualPath);
+		// Clean up
+		FileUtils.deleteDirectory(destinationFolder);
+	}	
+	
+	/**
+	 * @see ModuleUtil#expandJar(File,File,String,boolean)
+	 */
+	@Test
+	@Verifies(value = "expand directory without parent tree if name is directory and keepFullPath is false", method = "expandJar(File,File,String,boolean)")
+	public void expandJar_shouldExpandDirectoryWithoutParentTreeIfNameIsDirectoryAndKeepFullPathIsFalse() throws IOException {
+		// Arrange
+		final int numberOfFilesInSpecifiedDirectory = 2;
+		String directoryPath = "META-INF/maven/org.openmrs.module/test1-api";
+		File destinationFolder = this.getEmptyJarDestinationFolder();
+		// Act
+		ModuleUtil.expandJar(getJarFile(), destinationFolder, directoryPath, false);
+		// Assert
+		Object[] filesList = FileUtils.listFiles(destinationFolder, null, true).toArray();
+		Assert.assertEquals(numberOfFilesInSpecifiedDirectory, filesList.length);
+		String expectedPath = destinationFolder.getPath() + File.separator;
+		String ActualPath = ((File)filesList[0]).getPath().replace(((File)filesList[0]).getName(), "");
+		Assert.assertEquals(expectedPath, ActualPath);
+		// Clean up
+		FileUtils.deleteDirectory(destinationFolder);
```

Comment:
i think you can simplify your tests by leveraging the java `File` you dont need to use/add/replace file separators.

I think you can do something like

```java
	public void expandJar_shouldExpandDirectoryWithoutParentTreeIfNameIsDirectoryAndKeepFullPathIsFalse() throws IOException {

		final int numberOfFilesInSpecifiedDirectory = 2;
		String directoryPath = "META-INF/maven/org.openmrs.module/test1-api";
		File destinationFolder = this.getEmptyJarDestinationFolder();

		ModuleUtil.expandJar(getJarFile(), destinationFolder, directoryPath, false);

		List<File> actualExpandedFiles = (List<File>) FileUtils.listFiles(destinationFolder, null, true);
		Assert.assertEquals(numberOfFilesInSpecifiedDirectory, actualExpandedFiles.size());
		Assert.assertEquals(destinationFolder.toString(), actualExpandedFiles.get(0).getParent());

		FileUtils.deleteDirectory(destinationFolder);
	}
```